### PR TITLE
[mchehab-zbar] Add new port

### PR DESCRIPTION
**Pull request by Ulysses1337**
See issue #12510 and discussion in PR #15976. Haven't tried to build ARM myself.

### CODE REVIEW

File: ports/zbar-mchehab/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mchehab/zbar
+    REF 0.23.1
+    SHA512 ae7741cf750a10cf53dc11abcd482c3885507153ee37f6e3364ed5ed72184ebb009560b8c40d8090603a551fb681700a962838a59ce77d005d080ee49fbfa54b
+    HEAD_REF master
+    PATCHES
+        c99.patch
+        windows.patch
+        x64.patch
```
Comment (NancyLi1013) [inline ports/zbar-mchehab/portfile.cmake:14]: Can you submit these patches to upstream?

File: scripts/cmake/vcpkg_acquire_msys.cmake [hunk 2]
```diff
@@ -261,8 +261,13 @@ function(vcpkg_acquire_msys PATH_TO_ROOT_OUT)
         URL "https://repo.msys2.org/msys/x86_64/gmp-6.2.0-1-x86_64.pkg.tar.xz"
         SHA512 1389a443e775bb255d905665dd577bef7ed71d51a8c24d118097f8119c08c4dfe67505e88ddd1e9a3764dd1d50ed8b84fa34abefa797d257e90586f0cbf54de8
     )
-    msys_package( 
-        URL "https://repo.msys2.org/msys/x86_64/xz-5.2.5-1-x86_64.pkg.tar.xz" # this seems to require immediate updating on version bumps. 
+    msys_package(
+        URL "https://repo.msys2.org/msys/x86_64/tar-1.32-1-x86_64.pkg.tar.xz"
+        SHA512 379525f4b8a3f21d67d6506647aec8367724e1b4c896039f46845d9e834298280381e7261a87440925ee712794d43074f4ffb5e09e67a5195af810bbc107ad9a
+        DEPS libiconv libintl msys2-runtime
```
Comment (NancyLi1013) [inline scripts/cmake/vcpkg_acquire_msys.cmake:267]: Why do you add this part? Can you make a simple description here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @Ulysses1337 

Thanks for your PR to add this new port.

Could you please sign CLA first?

Seems it doesn't support uwp currently. Please help check it again. If yes, please update `supports `field.

File: ports/zbar-mchehab/portfile.cmake (hunk 1 above)
Comment (Ulysses1337) [reply ports/zbar-mchehab/portfile.cmake:14]: I'm not sure they'd appreciate ripping out some C99 syntax for the sake of a nonconformant compiler they don't primarily target.

The rest I'm confident would be acceptable, but my goal was to get a working port ASAP. Will this delay the port submission?

File: scripts/cmake/vcpkg_acquire_msys.cmake (hunk 2 above)
Comment (Ulysses1337) [reply scripts/cmake/vcpkg_acquire_msys.cmake:267]: The wonderful Autotools seem to rely on tar to get started. This was the only reasonable solution I saw.

[autoconf-x64-windows-err.log](https://github.com/microsoft/vcpkg/files/5950259/autoconf-x64-windows-err.log)

**Comment by Ulysses1337** [comment]
Oh I see, the library has some direct dependencies on the classic WinAPI. I edited the manifest to disable UWP. If anyone needs support in the future, it doesn't look hard to refactor.

I don't understand the x86 failure in colmap with lapack. It looks completely unrelated to my changes.

I figured I'd leave the agreement for last. BTW the _You have signed the CLA already_ line is quite misleading and should be fixed.

**Comment by Ulysses1337** [comment]
Not sure why people keep rerunning CI without new commits and while the code review discussion is sitting dead.

Amazingly, in the last few days mchehab has made ZBar work with MinGW and released a new version with Windows binaries as well. Upstream work is already underway to accommodate MSVC's woeful C99 support. I'll revise my submission for the latest stable release then, whatever level of MSVC support it happens to include.

File: ports/zbar-mchehab/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [reply ports/zbar-mchehab/portfile.cmake:14]: Sorry for the delay. It'd be great if upstream can accept the changes. If so, we can update the latest changes to this PR.

**Comment by NancyLi1013** [comment]
@Ulysses1337 

> Not sure why people keep rerunning CI without new commits and while the code review discussion is sitting dead.

We will rerun CI since there are some regressions that are not related with this PR.

> Amazingly, in the last few days mchehab has made ZBar work with MinGW and released a new version with Windows binaries as well. Upstream work is already underway to accommodate MSVC's woeful C99 support. I'll revise my submission for the latest stable release then, whatever level of MSVC support it happens to include.

So glad to hear the good news that upstream will support C99. Let's wait for the progress on upstream. You can remove the patches and update it to the new changes once all these parts work have been done by upstream.

**Comment by NancyLi1013** [comment]
Converting this PR to draft now until all work has been done.

**Comment by NancyLi1013** [comment]
Hi @Ulysses1337 

There has been the new version released in upstream. https://github.com/mchehab/zbar/releases

Could you please try to update to the latest version for adding this port?

**Comment by Ulysses1337** [comment]
I'm aware of the minor releases from Feb. They don't include the changes discussed above.

**Comment by Cheney-W** [comment]
Upstream has not released a new version, so let's use patch to fix the corresponding problem. Would you like to deal with the conflict here and continue to work on this PR?

**Comment by Ulysses1337** [comment]
Yes I'll work on it.

**Comment by Cheney-W** [comment]
Could you please handle the conflicts? Thank you!

**Comment by Cheney-W** [comment]
Pinging @Ulysses1337 for response. Is work still being done for this PR?

**Comment by Ulysses1337** [comment]
Yes.

File: ports/zbar-mchehab/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "zbar-mchehab",
+  "version": "0.23.1",
+  "port-version": 1,
```
Comment (Cheney-W) [inline ports/zbar-mchehab/vcpkg.json:4]:
```suggestion
```
New port should not have this field.

**Comment by Cheney-W** [comment]
Please add double quotes to the path string in portfile.cmake file.

File: ports/zbar-mchehab/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        ${CURRENT_PACKAGES_DIR}/debug/bin)
+endif()
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug/include
+    ${CURRENT_PACKAGES_DIR}/debug/share
+    ${CURRENT_PACKAGES_DIR}/share/doc)
+
+file(RENAME ${CURRENT_PACKAGES_DIR}/share/zbar ${CURRENT_PACKAGES_DIR}/share/${PORT})
+configure_file(${SOURCE_PATH}/COPYING ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
```
Comment (Cheney-W) [inline ports/zbar-mchehab/portfile.cmake:49]:
```suggestion
        "${CURRENT_PACKAGES_DIR}/bin"
        "${CURRENT_PACKAGES_DIR}/debug/bin")
endif()

file(REMOVE_RECURSE
    "${CURRENT_PACKAGES_DIR}/debug/include"
    "${CURRENT_PACKAGES_DIR}/debug/share"
    "${CURRENT_PACKAGES_DIR}/share/doc")

file(RENAME "${CURRENT_PACKAGES_DIR}/share/zbar" "${CURRENT_PACKAGES_DIR}/share/${PORT}")
configure_file("${SOURCE_PATH}/COPYING" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
```

Comment (Ulysses1337) [reply ports/zbar-mchehab/portfile.cmake:49]: Ah, all paths.

File: ports/zbar-mchehab/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "zbar-mchehab",
+  "version": "0.23.1",
+  "description": "ZBar is an open source software suite for reading bar codes from various sources, including webcams. This fork is actively maintained.",
+  "homepage": "https://github.com/mchehab/zbar",
+  "license": "LGPL-2.1-or-later",
```
Comment (JavierMatosD) [inline ports/zbar-mchehab/vcpkg.json:6]:
```suggestion
  "license": "LGPL-2.1",
```

File: ports/zbar-mchehab/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "zbar-mchehab",
+  "version": "0.23.1",
```
Comment (JavierMatosD) [inline ports/zbar-mchehab/vcpkg.json:3]: It seems the latest version for this repo is 0.23.90.

File: ports/zbar-mchehab/portfile.cmake (hunk 1 above)
Comment (JavierMatosD) [reply ports/zbar-mchehab/portfile.cmake:16]: Has upstream approved these changes?

File: ports/zbar-mchehab/vcpkg.json (hunk 5 above)
Comment (Ulysses1337) [reply ports/zbar-mchehab/vcpkg.json:6]: https://github.com/mchehab/zbar/blob/master/zbar/config.c
```
/*------------------------------------------------------------------------
 *  Copyright 2008-2010 (c) Jeff Brown <spadix@users.sourceforge.net>
 *
 *  This file is part of the ZBar Bar Code Reader.
 *
 *  The ZBar Bar Code Reader is free software; you can redistribute it
 *  and/or modify it under the terms of the GNU Lesser Public License as
 *  published by the Free Software Foundation; either version 2.1 of
 *  the License, or (at your option) any later version.
```

File: ports/zbar-mchehab/vcpkg.json (hunk 6 above)
Comment (Ulysses1337) [reply ports/zbar-mchehab/vcpkg.json:3]: Hmph. It was hidden from me and my Git client.
https://github.com/mchehab/zbar/tree/0.23.90
> This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

Comment (Ulysses1337) [reply ports/zbar-mchehab/vcpkg.json]: Well it has one little fix.

**Comment by dg0yt** [comment]
I'm concerned about the port name. We have this guideline: 
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#check-names-against-other-repositories 
so the name should probably be `mchehab-zbar`.

**Comment by Cheney-W** [comment]
@Ulysses1337 Could you please modify the port name as suggested by dg0yt?

**Comment by Ulysses1337** [comment]
As noted in an earlier discussion, my way makes it easier to find outside of the vcpkg search function. Also if we ever have other flavors, they'll all sort together this way. The zlib-ng port is kind of like this. Does anyone else want to put in their two cents?

**Comment by dg0yt** [comment]
I did understand the intention of the name, but vcpkg decided to have a different guideline. 
Port `zlib-ng` is for a repo which is actually named `zlib-ng`, so it doesn't offer a counter-example.

**Comment by Ulysses1337** [comment]
I was careful to say kind of. Still its name shows the convention's handiness. I didn't find any ports of plain forks of known libraries which need to avoid confusion. Perhaps the guideline should change. Does anyone else want to chime in?

**Comment by Neumann-A** [comment]
the `shiftmedia` stuff is also prefixed with it instead of suffixed

**Comment by Ulysses1337** [comment]
Ah, that's brand new so I didn't see it.

**Comment by Ulysses1337** [comment]
Oddly, shiftmedia still isn't listed on vcpkg.io.

Since the discussion has died out, I guess it's time.

**Comment by Neumann-A** [comment]
vcpkg.io needs manual updating.

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**